scripts/fig1b-modality_barchart.R

```diff
@@ -0,0 +1,138 @@
+
+# read in library metadata and filter to only projects in whitelist
+library_metadata_df <- readr::read_tsv(library_metadata_file) |> 
+  dplyr::filter(scpca_project_id %in% project_whitelist) |>
+  dplyr::select(scpca_project_id, scpca_sample_id, scpca_library_id, seq_unit, technology) |> 
+  # create a modality column that labels everything as single suspension, bulk, spatial, or CITE
+  dplyr::mutate(
+    modality = dplyr::case_when(
+      seq_unit == "bulk" ~ "Bulk",
+      seq_unit == "spot" ~ "Spatial transcriptomics",
+      stringr::str_detect(technology,"CITE") ~ "With CITE-seq",
+      stringr::str_detect(technology, "cellhash") ~ "With cell hashing",
+      seq_unit %in% c("cell", "nucleus") ~ "Single suspension"
+    )
+  )
+
+# get a list of all bulk and spatial samples
+# these will be checked against all single-cell/ nuc samples to look for matching samples
+all_bulk <- library_metadata_df |> 
+  dplyr::filter(modality == "Bulk") |> 
+  dplyr::pull(scpca_sample_id)
+
+all_spatial <- library_metadata_df |> 
+  dplyr::filter(modality == "Spatial transcriptomics") |>
+  dplyr::pull(scpca_sample_id)
+
+# get list of all single cell and all single nuclei
+all_single_cell <- library_metadata_df |> 
+  dplyr::filter(seq_unit == "cell" & modality == "Single suspension") |> 
+  dplyr::pull(scpca_sample_id)
+all_single_nuc <- library_metadata_df |> 
+  dplyr::filter(seq_unit == "nucleus" & modality == "Single suspension") |> 
+  dplyr::pull(scpca_sample_id)
+
+# get list of all bulk and spatial samples that don't have a corresponding cell/nucleus sample
+all_suspension <- c(all_single_cell, all_single_nuc)
+bulk_only <- setdiff(all_bulk, all_suspension)
+spatial_only <- setdiff(all_spatial, all_suspension)
+
+
+# remove bulk and spatial only from metadata
+filtered_modality_df <- library_metadata_df |> 
+  dplyr::filter(!(scpca_sample_id %in% c(bulk_only, spatial_only))) |>
+  # make sure all bulk and spatial get designated with cell or nucleus
+  # also rename to be more specific when creating legend
+  dplyr::mutate(
+    seq_unit = dplyr::case_when(
+      scpca_sample_id %in% all_single_cell ~ "Single-cell",
+      scpca_sample_id %in% all_single_nuc ~ "Single-nuclei"
+    )
+  ) |> 
+  # first combine all modalities for each sample id into one list
+  # make sure to keep seq unit
+  dplyr::group_by(scpca_sample_id, seq_unit) |>
+  dplyr::summarise(modality = paste(modality, collapse = ";")) |> 
+  # split any lists of modalities so we have one row per sample per modality
+  dplyr::mutate(modality = stringr::str_split(modality, ";")) |> 
+  tidyr::unnest(modality) |>
+  dplyr::group_by(modality) |> 
+  # get the total for each modality to use for specifying order in plot
+  dplyr::add_tally(name = "total_per_modality") |>
+  # add a column to help pull out additional modalities into its own facet 
+  dplyr::mutate(additional_mods = ifelse(modality == "Single suspension", "All samples", "Samples with additional modalities"),
```

Comment:
Can you generally update the spacing here, with `additional_mods =` on new line, with `ifelse` maybe spanning a couple lines?